tracker/src/main/kotlin/org/dhis2/tracker/relationships/ui/RelationshipsScreen.kt

```diff
@@ -0,0 +1,306 @@
+import org.hisp.dhis.mobile.ui.designsystem.component.ProgressIndicator
+import org.hisp.dhis.mobile.ui.designsystem.component.ProgressIndicatorType
+import org.hisp.dhis.mobile.ui.designsystem.component.Title
+import org.hisp.dhis.mobile.ui.designsystem.component.state.rememberAdditionalInfoColumnState
+import org.hisp.dhis.mobile.ui.designsystem.component.state.rememberListCardState
+import org.hisp.dhis.mobile.ui.designsystem.theme.Shape
+import org.hisp.dhis.mobile.ui.designsystem.theme.Spacing
+import org.hisp.dhis.mobile.ui.designsystem.theme.TextColor
+
+@Composable
+fun RelationShipsScreen(
+    uiState: RelationshipsUiState<List<RelationshipSection>>,
+    onCreateRelationshipClick: (RelationshipSection) -> Unit,
+    onRelationshipClick: (RelationshipItem) -> Unit,
+) {
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .background(color = Color.White, shape = Shape.LargeTop)
+            .padding(horizontal = Spacing.Spacing8),
+        verticalArrangement = spacedBy(Spacing.Spacing4),
+    ) {
+        when (uiState) {
+            is RelationshipsUiState.Loading -> {
+                Box(
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .weight(1f),
+                    contentAlignment = Alignment.Center,
+                ) {
+                    ProgressIndicator(type = ProgressIndicatorType.CIRCULAR)
+                }
+            }
+
+            is RelationshipsUiState.Empty,
+            is RelationshipsUiState.Error -> NoRelationships()
+
+            is RelationshipsUiState.Success -> {
+                LazyColumn {
+                    items(uiState.data) { item ->
+                        RelationShipTypeSection(
+                            title = item.relationshipType.displayName() ?: "",
+                            description = if (item.relationships.isEmpty()) "No data" else null,
+                            relationships = item.relationships,
+                            canAddRelationship = item.canAddRelationship(),
+                            onCreateRelationshipClick = {
+                                onCreateRelationshipClick(item)
+                            },
+                            onRelationshipClick = {
+                                onRelationshipClick(it)
+                            }
+                        )
+                    }
+                }
+            }
+        }
+    }
+}
+
+@Composable
+private fun RelationShipTypeSection(
+    modifier: Modifier = Modifier,
+    title: String,
+    description: String?,
+    relationships: List<RelationshipItem>,
+    canAddRelationship: Boolean,
+    onCreateRelationshipClick: () -> Unit,
+    onRelationshipClick: (RelationshipItem) -> Unit,
+) {
+    Column(
+        modifier = modifier
+            .fillMaxWidth()
+            .wrapContentHeight()
+            .padding(16.dp),
+        verticalArrangement = spacedBy(4.dp),
```

Comment:
Could you replace the .dp paddings and spacings with mobile-ui's spacing object?